### DISCUSSION TITLE
Backup function within settings/tools

### DISCUSSION THREAD

**Discussion by Burki24** [Koenkk/zigbee2mqtt#12731] [General]
@Koenkk, @nurikk 
Actually, it is possible to download the state.json. For having a backup, it`s not enough. So my question:

will it be possible to place a second button with "download complete data", which will download a zip-package of the complete "data"-folder as a backup (Preferably with the download date and time in the file name.) ?

greetings
Burkhard

**Comment by nurikk** [comment]
`state.json` from frontend is runtime state of [React](https://reactjs.org/) application, not z2m.

**Reply by Burki24** [reply]
> `state.json` from frontend is runtime state of [React](https://reactjs.org/) application, not z2m.

Okay, but will it be possible to integrate a backup-button for the data folder?
Will make a backup quite easier for some users.

**Reply by nurikk** [reply]
Yep, this would be nice. I hope someone will implement corresponding api in z2m

**Reply by Koenkk** [reply]
Should be easy to implement, what response would be the easiest, a binary blob of the zip?

**Reply by nurikk** [reply]
> Should be easy to implement, what response would be the easiest, a binary blob of the zip?

It would be nice to have two http api
1) download file z2mbackup.yyyymmdd.zip
2) upload backup file and restart after upload

**Reply by Koenkk** [reply]
@nurikk can you check https://github.com/Koenkk/zigbee2mqtt/pull/12797 ?

**Reply by nurikk** [reply]
> @nurikk can you check #12797 ?

This implementation will ignore all sub folders (ex: custom user extensions and maybe external converters) which will make backup incomplete

**Comment by onedr0p** [comment]
Hi 👋🏼 @Koenkk 

Would it be possible extend the current implementation to have backups automated and stored on disk? The below pseudo-config would backup once a day and only retain 7 days of backups and the zips would be stored in the /config/backups directory. This would be very helpful at looking at configuration drift and correcting mistakes by looking at old configs. Thanks!

```yaml
backup_interval: 1d
backup_retain: 7d
backup_path: /config/backups
```

This would be especially help in a container install of z2m where automating this outside the z2m container is more difficult.

**Reply by Koenkk** [reply]
@onedr0p please make a feature request so I can monitor interest for this. Keep in mind that a daily backup is already made.

**Reply by onedr0p** [reply]
Issue created https://github.com/Koenkk/zigbee2mqtt/issues/13017

**Comment by julianfs** [comment]
I concur with this issue, I'm also seeing the same problem with 2 (so far) of my 4 devices.

Those that lost connectivity I moved to ZHA on the same device and they've been fine.

Z2M 1.26.0, Conbee II (2 separate ones for ZHA & Z2M) on a Pi4.
